--- device.c.steerable	2005-01-09 18:09:19.996328466 +0100
+++ device.c	2005-01-09 18:10:10.572853731 +0100
@@ -1057,10 +1057,12 @@
      return false;
   if (Receiver->device == this)
      return true;
+  /*   
   if (!HasLock(TUNER_LOCK_TIMEOUT)) {
      esyslog("ERROR: device %d has no lock, can't attach receiver!", CardIndex() + 1);
      return false;
      }
+  */   
   cMutexLock MutexLock(&mutexReceiver);
   for (int i = 0; i < MAXRECEIVERS; i++) {
       if (!receiver[i]) {
--- recorder.c.steerable	2005-01-09 18:12:09.844359502 +0100
+++ recorder.c	2005-01-09 18:13:37.276606354 +0100
@@ -96,11 +96,14 @@
 void cFileWriter::Action(void)
 {
   time_t t = time(NULL);
+  time_t m = MAXBROKENTIMEOUT * 10;  //give more time to first packet
+   
   active = true;
   while (active) {
         int Count;
         uchar *p = remux->Get(Count, &pictureType);
         if (p) {
+           m = MAXBROKENTIMEOUT;
            //XXX+ active??? see old version (Busy)
            if (!active && pictureType == I_FRAME) // finish the recording before the next 'I' frame
               break;
@@ -118,7 +121,7 @@
               break;
            t = time(NULL);
            }
-        else if (time(NULL) - t > MAXBROKENTIMEOUT) {
+        else if (time(NULL) - t > m) {
            esyslog("ERROR: video data stream broken");
            cThread::EmergencyExit(true);
            t = time(NULL);
