--- module/actuator.c.orig	2004-08-31 15:16:07.000000000 +0400
+++ module/actuator.c	2006-02-17 22:45:11.000000000 +0300
@@ -68,7 +68,7 @@
 
 struct parport *pport;
 struct pardevice *ppdevice;
-struct class_simple *actuator_class;
+struct class *actuator_class;
 
 /* timer, used as a delay for position reached/direction changed */
 
@@ -356,8 +356,8 @@
 #ifdef CONFIG_DEVFS_FS
      devfs_mk_cdev(MKDEV(major, 0), S_IFCHR | S_IRUSR | S_IWUSR, DRIVER_NAME);
 #endif     
-     actuator_class = class_simple_create(THIS_MODULE, DRIVER_NAME);
-     class_simple_device_add(actuator_class,MKDEV(major,0),NULL,DRIVER_NAME);
+     actuator_class = class_create(THIS_MODULE, DRIVER_NAME);
+     class_device_create(actuator_class,MKDEV(major,0),NULL,DRIVER_NAME);
 
     /* set up timer function */
     init_timer(&timer);
@@ -402,8 +402,8 @@
 #ifdef CONFIG_DEVFS_FS
     devfs_remove(DRIVER_NAME);
 #endif    
-    class_simple_device_remove(MKDEV(major,0));
-    class_simple_destroy(actuator_class);
+    class_device_destroy(actuator_class,MKDEV(major,0));
+    class_destroy(actuator_class);
     unregister_chrdev(major, DRIVER_NAME); 
 
     /* release the parallel port */
